### ISSUE TITLE
Speed up with touch control

### ISSUE THREAD

**Issue by Rockbard** [TeamNewPipe/NewPipe#1823]
**Feature request:**
 Speed up with touch control instead of brightness control (as an option)
 **Purpose:**
 Watching  tutorials on youtube will be great if you could quickly speed up video and turn on a normal play speed when needed with convenient touch control
 **Note**
 Volume pitch should be at 100%

Thanks,
Dmitry

**Comment by bkqtnte10** [comment]
What if the speed setting is with swiping left to right (instead up to down)?

**Comment by Rockbard** [comment]
Right to left is filling better as time search function. That's a pretty a
match standard for all players.
 In my opinion brightness with the up and down gesture is arguable, as I
accidentally touching it all the time. But that a standard gesture for most
of the players too.

 Maybe (just an idea) speed is up and down gesture on the left, brightness
in the middle and volume on the right side?

**Comment by TobiGr** [comment]
IMO, three gestures are too much. But @ritiek made it possible in #1644 to disable either the volume or brightness control or even both. So if someone implements a third logic which handles the speed and restricts the maximum number of enabled gestures to two, I'd be okay with this.
@theScrabi @ritiek What do you think?

**Comment by ritiek** [comment]
This should be a good addition. The only caveat I can think of is that three gestures enabled all at once might cause some trouble in triggering the correct behaviour but this shouldn't be a problem as one would be able to disable the gestures they don't need.

> Maybe (just an idea) speed is up and down gesture on the left, brightness in the middle and volume on the right side?

Just my $0.02. IMO as @bkqtnte10 mentioned it would be nicer to have the speed gesture as swiping left-right on any part of the screen. Currently, swiping top-down on the left part of screen affects the brightness and on the right part of screen affects the volume. In the same way, swiping left-right on any part of the screen would affect the playback speed.

In this way, we would be able to have all three gestures (brightness, volume, playback speed) enabled at the same time.

**Comment by Tikotik** [comment]
If someone is looking into this, #1536 is related.
Remember playback speed changes between videos.

**Comment by androidSwDev099** [comment]
How about left/right gesture on top half of the screen to control video play speed/tempo and on left/right gesture on bottom half of the screen for rewind/forward video time?

**Comment by dardoor** [comment]
There's also the double finger gesture, like in MX Player, for example. This way you can easily have three gestures without confusion (in MX Player there's also a zoom gesture, which does get in the way, but you can disable it in the settings).

Personally, I change speed more often than brightness, so I'd relegate the brightness gesture to "two fingers" and make speed changing easier by having it on the left in place of the current brightness gesture.

**Comment by MD77MD** [comment]
from reading all the comments, my suggestion is to have gestures as fallow
1. right to left for seek 
2.  up/down on the left for volume
3. up/down on the middle for speed
4. up/down on the left for brightness

the up/down gestures should fallow the three double tab zones.. this way although there are three active up/down  gestures, they are clearly distinguished by the three zones that the users are already familiar with.

**Comment by dardoor** [comment]
> but to tell you the truth I don't think this feature is a most as the speed button is close enough for fast changes.

I disagree. I think even just copying the MX Player gesture would make it significantly faster and nicer.

Three zones could work too, but it would probably be too crowded at that point and lead to mistakes.

**Comment by EliasKomar** [comment]
I think the whole where to put what debate is useless, just implement 3 gestures up-down-left, up-down-right and left-right and also the code to change playback speed and playback position and let the user pick how they want to connect them, just like android gestures for lock screen.

**Comment by SameenAhnaf** [comment]
Why don't we use MX Player gesture for playback speed?? I have been user for years and it's not confusing at all.

When a video is playing, use two fingers to tap and hold on the screen. You can pause the video. Drag those two fingers towards the top to increase the playback speed gradually. Drag them downwards to decrease the playback speed. A percentage value will be shown on the screen.

**Comment by SameenAhnaf** [comment]
I suggested a better solution in #5821. There are already way too many gestures. So, it's better to access it from "Always-on button". "Playback speed" is also an option and certain options should already stay selected to switch speeds easily.

**Comment by bkqtnte10** [comment]
Maybe it could be implemented like this:

-     tap and hold right side of screen increases tempo by +75%, after release back to previous tempo
-     tap and hold left side of screen decreases tempo by -75%, after release back to previous tempo

(perhaps additionally change the tempo if the finger is moved further left/right.

**Comment by fritzrehde** [comment]
Any progress on this?

**Comment by Rockbard** [comment]
Let's keep the two areas as they are now, but allow users to choose between Volume, Brightness, and Speed on both sides.

Speed adjustment could resemble Brightness and Volume, with the radial line moving along the circle.

As an example, I currently use Volume Adjust on both sides. If there were a Speed Adjusting option to choose, I would definitely use it for my left adjustment area.

**Comment by timur-g** [comment]
I like "left/right gesture on top half of the screen to control video play speed/tempo". Very desirable.

**Comment by goyalyashpal** [comment]
this would help a lot while watching studies related classroom-lectures too.

say where the teacher is just writing something he has already spoken, or talking some off-topic; but u can't skip ahead lest miss something.

so speed-up/down gesture would be a boon.

-----
i would strongly disagree with the "niche" label _"Only relevant to a very small amount of people"_.  online education on youtube is billions of dollar industry, aint no niche.
